apps/wfo-ui-surf/components/WfoServiceTicketDetailPage/WfoServiceTicketSentEmails.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from 'react';
+
+import { useTranslations } from 'next-intl';
+
+import { EuiSpacer, EuiText } from '@elastic/eui';
+
+import { WfoEmailList } from '@/components/WfoEmailList/WfoEmailList';
+import {
+    EmailListItem,
+    EmailLog,
+    EmailStep,
+    ServiceTicketLogType,
+    ServiceTicketProcessState,
+    ServiceTicketWithDetails,
+} from '@/types';
+
+interface WfoSubscriptionGeneralProps {
+    serviceTicketDetail: ServiceTicketWithDetails;
+}
+
+const mapLogEntryToStep = (emailLog: EmailLog): EmailStep => ({
+    name: emailLog.name,
+    status: emailLog.log_type as ServiceTicketLogType,
+    sentBy: emailLog.sentBy,
+    executed: new Date(emailLog.entry_time).toISOString(),
+    stepId: emailLog.log_id,
+    state: {
+        parameter1: 'example',
+        parameter2: 56,
+        parameter3: false,
+    },
+    stateDelta: {
+        parameter1: 'example',
+    },
+});
+
+export const WfoServiceTicketSentEmails = ({
+    serviceTicketDetail,
+}: WfoSubscriptionGeneralProps) => {
+    const t = useTranslations(
+        'cim.serviceTickets.detail.tabDetails.sentEmails',
+    );
+
+    // Add sentBy to email logs and UPDATE prefixes to update emails
+    const data = serviceTicketDetail.email_logs
+        .sort((a, b) => {
+            const timeA = new Date(a.entry_time).getTime();
+            const timeB = new Date(b.entry_time).getTime();
+            return timeA - timeB;
+        })
+        .map((sent_email, index, array) => {
+            const match = serviceTicketDetail.logs.find(
+                (log) => log.log_id === sent_email.log_id,
+            );
+            const allUpdateEmails = array.filter(
+                (email) => email.log_type === ServiceTicketLogType.UPDATE,
+            );
+            const slicedUpdateEmails = allUpdateEmails.slice(index);
+
+            return {
+                ...sent_email,
+                sentBy: match ? match.logged_by : '',
+                name:
+                    sent_email.log_type === ServiceTicketLogType.UPDATE
```

Comment:
I think it would be better to move this into a function then multiple ternary operators.